lib/uuid_mixin.rb

```diff
@@ -1,17 +1,12 @@
 module UuidMixin
   extend ActiveSupport::Concern
   included do
-    before_validation :set_guid, :on => :create if respond_to?(:before_validation)
+    default_value_for(:guid) { SecureRandom.uuid }
```

Comment:
2 questions

- Is this sufficient for ensuring a GUID is always created?  What happens if the user creates an object with `:guid => nil` explicitly?
- The `if respond_to?(:before_validation)` was because UuidMixin was used in non-Model objects (I think ActsAsArModel).  Is this not the case anymore?

Reply:
> The `if respond_to?(:before_validation)` was because UuidMixin was used in non-Model objects (I think ActsAsArModel).  Is this not the case anymore

?That doesn't appear to be the case anymore
```ruby
irb(main):005:0> ActsAsArModel.descendants.select { |i| i.ancestors.include?(UuidMixin) }
=> []
```

Reply:
> What happens if the user creates an object with `:guid => nil` explicitly?

Then it would be `nil`

Reply:
I don't think any of these classes allow the user to set the `guid`

```ruby
irb(main):009:0> ApplicationRecord.descendants.select { |i| i.ancestors.include?(UuidMixin) }.collect { |i| i.base_class.name }.uniq
=> ["MiqServer", "MiqRegion", "MiqWorker", "AutomateWorkspace", "ConditionSet", "Condition", "Job", "LifecycleEvent", "ScanItem", "CustomButtonSet", "CustomButton", "ExtManagementSystem", "Host", "VmOrTemplate", "Service", "MiqAction", "MiqActionSet", "MiqAeWorkspace", "MiqAlertSet", "MiqAlert", "MiqEventDefinition", "MiqEventDefinitionSet", "MiqPolicySet", "MiqPolicy", "MiqWidgetSet", "MiqWidget", "ScanItemSet", "ChargebackRate", "ServiceTemplate", "Provider"]
```